koku/forecast/forecast.py

```diff
@@ -178,12 +183,19 @@ def _predict(self, data):
         if len(data) < self.MINIMUM:
             LOG.warning("Number of data elements is fewer than the minimum.")
 
-        # arrange the data into a form that statsmodels will accept.
         dates, costs = zip(*data)
-        X = [int(d.strftime("%Y%m%d")) for d in dates]
+
+        # format the independent variable so that all points are equidistant.
+        # in this case, every point is 86400.0 seconds apart.
+        X = []
+        for d in dates:
+            X.append(datetime(d.year, d.month, d.day, 0, 0, 0, 0).timestamp())
```

Comment:
@esebesto made a suggestion in Jira that this could be just simple small integers.  Have you been able to give that a try?  It would be easier to work with while testing/debugging.

The comment also had some great use cases to try regarding whether or not this PR would address the missing constant.

Reply:
We're discussing that stuff outside of github. :slightly_smiling_face: